### PR TITLE
Allow a trailing comma in lint_array

### LINKED ISSUE #47428: Support trailing commas in lint_array
https://github.com/rust-lang-nursery/rust-clippy/pull/2350#discussion_r161404874

### PR DESCRIPTION
**Pull request by mark-i-m**
fix #47428

### CODE REVIEW

File: src/librustc/lint/mod.rs [hunk 1]
```diff
@@ -97,12 +97,13 @@ macro_rules! declare_lint {
...
 /// Declare a static `LintArray` and return it as an expression.
 #[macro_export]
-macro_rules! lint_array { ($( $lint:expr ),*) => (
-    {
-        static ARRAY: LintArray = &[ $( &$lint ),* ];
-        ARRAY
-    }
-) }
+macro_rules! lint_array {
+    ($( $lint:expr ),*,) => { lint_array!( $( &$lint ),* ) };
```
Comment (durka) [inline src/librustc/lint/mod.rs]: Why did you add a `&` here?

Comment (mark-i-m) [reply src/librustc/lint/mod.rs]: Oh, good catch... Copy/paste :face_with_head_bandage:

Comment (mark-i-m) [reply src/librustc/lint/mod.rs]: Fixed. Thanks @durka !

**Comment by durka** [comment]
LGTM, I'm not a reviewer though.

**Comment by mark-i-m** [comment]
Thanks, I added a couple of tests...

**Comment by mark-i-m** [comment]
Hmm... ok... I need to fix a couple of errors.

**Comment by mark-i-m** [comment]
Ok, I think they should actually work now

**Comment by mark-i-m** [comment]
@estebank ping :)

**Comment by estebank** [comment]
@mark-i-m one final nitpick and r=me. Could you squash all your commits into one?

**Comment by mark-i-m** [comment]
@estebank Squashed :)

Thanks!
